### ISSUE TITLE
Stepper and Stepper Item - Ensure custom controls provide proper textual name, role, and state information - (2036570226)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#5602]
### Module: 
 31 Stepper and Stepper Item 

### Violation: 
 Ensure custom controls provide proper textual name, role, and state information 

![image](https://user-images.githubusercontent.com/5023024/202740101-5bef66c5-e407-467f-9a4f-66b1c741d3f1.png)

### WCAG Reference: 
- [1.1.1 Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036570226 

### Severity: 
 7 

### Description: 
 [Issue]
The current step information is not communicated to JAWS users

[User Impact]
Screen reader users will have difficulty determining the purpose and state of these controls.

[Code Reference] 
 ```
<calcite-stepper-item item-title="Error property" error="" aria-expanded="true" layout="horizontal" scale="m" tabindex="0" calcite-hydrated="" active="">Step 2 contains an "error" property.</calcite-stepper-item>
 ``` 

### Note: 
 [Suggestion]
Ensure custom controls provide proper textual name, role, and state information. 

A control&#039;s state indicates several possible pieces of information. Common states include expanded, current, selected, checked, and disabled. Most are indicated with similarly named ARIA attributes. Certain states can only be used with certain roles. For a list of roles and their allowed states, see the ARIA specification: https://www.w3.org/TR/wai-aria/#role_definitions

A proper ordered list can also be used with a list item that includes every step and it&#039;s respective text. Visually hidden text can be used to indicate the current and completed steps. 

[Compliant Code Example] 
 ```
<calcite-stepper-item error="" aria-expanded="true" layout="horizontal" scale="m" tabindex="0" calcite-hydrated="" active="">
<span class="visuallyhidden">Current: </span>
Step 2 contains an "error" property.
</calcite-stepper-item>


 ```
[Recommended Reading]
More information on how to indicate progress, reference approach 4 in "Multi-page forms" best practices:
https://www.w3.org/WAI/tutorials/forms/multi-page/

More information about the correct use of ARIA states, roles and properties can be found on the Level Access website:

https://www.levelaccess.com/how-not-to-misuse-aria-states-properties-and-roles/ 

### Media Type: 
 Custom Controls 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
cc #5603

**Comment by geospatialem** [comment]
Verified on the `main` branch with NVDA (provides current step and completed step) and JAWS (provides current step).
